benches/eth_deploy_code.rs

```diff
@@ -0,0 +1,70 @@
+use criterion::{criterion_group, criterion_main, BatchSize, BenchmarkId, Criterion, Throughput};
+use secp256k1::SecretKey;
+
+mod common;
+
+use common::{address_from_secret_key, create_eth_transaction, deploy_evm, RAW_CALL};
+
+const INITIAL_BALANCE: u64 = 1000;
+const INITIAL_NONCE: u64 = 0;
+const TRANSFER_AMOUNT: u64 = 0;
+
+fn eth_deploy_code_benchmark(c: &mut Criterion) {
+    let mut runner = deploy_evm();
+    let mut rng = rand::thread_rng();
+    let source_account = SecretKey::random(&mut rng);
+    runner.create_address(
+        address_from_secret_key(&source_account),
+        INITIAL_BALANCE.into(),
+        INITIAL_NONCE.into(),
+    );
+    let inputs: Vec<_> = [1, 4, 8, 12, 16]
+        .iter()
+        .copied()
```

Comment:
Minor note, since these are digits, they don't need to be copied in this way as they can be de-referenced:

 ```rust
     let inputs: Vec<_> = [1, 4, 8, 12, 16]
        .iter()
        .map(|n| {
            let code_size = 2usize.pow(*n);
            let code: Vec<u8> = std::iter::repeat(0).take(code_size).collect();
            let transaction = create_eth_transaction(
                None,
                TRANSFER_AMOUNT.into(),
                code,
                Some(runner.chain_id),
                &source_account,
            );
            rlp::encode(&transaction).to_vec()
        })
        .collect();
```

Reply:
Dereferencing is equivalent to copying though right? I.e.

```rust
[...].iter().copied()
```

produces the same instructions on compilation as

```rust
[...].iter().map(|n| *n)
```

?

Reply:
There are three equivalent ways to write this (in order of my personal preference): 

* `iter().copied().map(|n| ...)` -- clearly factors copying out, so that it's not distracing
* `iter().map(|&n| ...)` -- dereference once in in a pattern
* `iter().map(|n| *n)` -- dereference every time n is used

Perf-wise, they are all equivalent.

Reply:
Thanks @matklad !